### ISSUE TITLE
[Linter] Rule configuration severity is not enforced

### ISSUE THREAD

**Issue by jbanety** [oxc-project/oxc#2059]
I encountered an issue with the configuration of the @typescript-eslint/no-explicit-any rule in the `eslintrc.json` file. Despite the rule being set to "error", warnings are still being generated instead of errors.

### Configuration
`eslintrc.json`
```json
{
  "rules": {
    "@typescript-eslint/no-explicit-any": "error"
  }
}
```

### Command line output
```shell
⚠ typescript-eslint(no-explicit-any): Unexpected any. Specify a different type.
     ╭─[src/helpers/htmlTemplate.ts:144:1]
 144 │ 
 145 │ export function renderHtml(template: string, data: Record<string, any> = {}): string {
     ·                                                                   ───
 146 │   // Render template
     ╰────
  help: Use `unknown` instead, this will force you to explicitly, and safely, assert the type is correct.

Finished in 20ms on 34 files with 83 rules using 16 threads.
Found 62 warnings and 0 errors.
```

PR https://github.com/oxc-project/oxc/pull/2031 fixed the `off` problem but not the severity.
Tested on the latest 0.2.1.

**Comment by jbanety** [comment]
I'm trying to do a PR to correct the issue.
It seems that the replace (here : https://github.com/oxc-project/oxc/blob/main/crates/oxc_linter/src/config/mod.rs#L97) makes the associated severity disappear. 
It's always Allow.

**Comment by jbanety** [comment]
Also, the `oxlint -D all` command only outputs warnings.

**Comment by jbanety** [comment]
From what I understand, severity is hardcoded into the rules.
So we can't change the gravity by configuration at the moment?

**Comment by jbanety** [comment]
@Boshen I can help on this one.
I have already looked a little at how we could overcome this problem but it affects the design of Oxlint.

**Comment by Boshen** [comment]
Changing a diagnostic to an error was never implemented. The current design treats everything as a warning.

I would consider it as a bug if vscode raises any errors at the moment.

---

We can also start a topic on changing the current implementation to distinguish warnings and errors when we supply "--deny" / "--warn" from the cli and "error" / "warn" from the config.

**Comment by alisnic** [comment]
The whole error/warning flow is currently broken from my tests. Given a `file.tsx` with the following contents:

```
var Hello = <a target="_blank" href="https://example.com/"></a>;
```

Running with `--deny-warnings` has no effect:

```
$ npx oxlint@latest --deny-warnings

  ⚠ eslint-plugin-react(jsx-no-target-blank): Using target=`_blank` without rel=`noreferrer` (which implies rel=`noopener`) is a security risk in older browsers: see https://mathiasbynens.github.io/rel-noopener/#recommendations
   ╭─[file.tsx:1:23]
 1 │ var Hello = <a target="_blank" href="https://example.com/"></a>;
   ·                       ────────
   ╰────
  help: add rel=`noreferrer` to the element

Finished in 2ms on 1 file with 86 rules using 10 threads.
Found 1 warning and 0 errors.

```

Running with `-D correctness` has also no effect:

```
$ npx oxlint@latest -D correctness

  ⚠ eslint-plugin-react(jsx-no-target-blank): Using target=`_blank` without rel=`noreferrer` (which implies rel=`noopener`) is a security risk in older browsers: see https://mathiasbynens.github.io/rel-noopener/#recommendations
   ╭─[file.tsx:1:23]
 1 │ var Hello = <a target="_blank" href="https://example.com/"></a>;
   ·                       ────────
   ╰────
  help: add rel=`noreferrer` to the element

Finished in 2ms on 1 file with 86 rules using 10 threads.
Found 1 warning and 0 errors.
```

No matter what I do, I can't make oxlint to return a non-zero status when running.

**Comment by Boshen** [comment]
@alisnic 

```bash
npx oxlint@latest --deny-warnings
echo $?
```

This is not printting `1`?

**Comment by alisnic** [comment]
It is printing, I am confused. I'm sorry, I guess I was confused by the "Found 1 warning" message

```
$ echo $?
1
```

**Comment by Boshen** [comment]
I can fix this now after changing how diagnostics work under the hook

https://github.com/oxc-project/oxc/blob/f6daf0b7ae4fb9a3ac9e8520824eb06a278c78ea/crates/oxc_diagnostics/src/lib.rs#L47

**Comment by Boshen** [comment]
Next version:

Breaking change:

* `--deny` / `-D` from the CLI and `error` from configuration file will report diagnostics as severity "error".
* `--warn` / `-W` from the CLI and `warn` from configuration file will report diagnostics as severity "warning".
